pkg/cmd/logs.go

```diff
@@ -0,0 +1,235 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+)
+
+type LogsOptions struct {
+	ContainerName string
+	Name          string
+	Resource      string
+	GroupVersion  string
+	LabelSelector string
+
+	Token string
+
+	KubeArchiveHost string
+	KubeArchiveCA   string
+
+	RESTConfig *rest.Config
+	kubeFlags  *genericclioptions.ConfigFlags
+}
+
+var logsLong = `Print the logs for a container in a pod or specified resource from KubeArchive.
+If the provided resource owns directly or indirectly a pod, and all the resources in the chain are archived
+you can also get the logs one of the containers in the chain (latest one if it is not specified).`
+
+var logsExample = `
+# Return logs from pod nginx with only one container
+kubectl archive logs nginx
+
+# Return logs from pod nginx by specifying the container name
+kubectl archive logs nginx -c my-container
+
+# Return logs from a resource different from a pod. It will pick one of the pod's logs.
```

Comment:
`It will pick one of the pod's logs` -> it will return the latest in this case, right?

Reply:
It depends on the API implementation, in this case yes. If we change the API the CLI won't have a say in it.